providers/oidc.go

```diff
@@ -285,21 +289,35 @@ func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface
 	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
 	if rawGroups != nil && ok {
 		for _, rawGroup := range rawGroups {
-			group, ok := rawGroup.(string)
-			if ok {
-				groups = append(groups, group)
+			formattedGroup := formatGroup(rawGroup)
+			if formattedGroup != "" {
+				groups = append(groups, formattedGroup)
 			}
 		}
 	}
 
 	return groups
+
+}
+
+func formatGroup(rawGroup interface{}) string {
+	group, ok := rawGroup.(string)
+	if !ok {
+		jsonGroup, err := json.Marshal(rawGroup)
+		if err != nil {
+			logger.Errorf("unable to format group of type %s with error %s", reflect.TypeOf(rawGroup), err)
+			return ""
```

Comment:
Last NIT and then I'm ready to merge -- sorry to make you go back on this formatting 😅 

I think now with this structure, this case should return an error instead just `""` --

then above the usage should switch to this:

```
			formattedGroup, err := formatGroup(rawGroup)
			if err != nil {
				// Potentially log here instead of in formatGroup?
				continue
			}
			groups = append(groups, formattedGroup)
```

Reply:
Makes more sense! 👍